test/language/expressions/class/class-name-ident-let-escaped.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2018 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-class-definitions
+description: >
+  `let` with escape sequence is not a valid class-name identifier.
+info: |
+  12.1.1 Static Semantics: Early Errors
+
+  Identifier : IdentifierName but not ReservedWord
+
+  It is a Syntax Error if this phrase is contained in strict mode code and the
+  StringValue of IdentifierName is: "implements", "interface", "let", "package",
+  "private", "protected", "public", "static", or "yield".
```

Comment:
I'd like to add this section: https://tc39.github.io/ecma262/#sec-strict-mode-code, but you don't have to do it if it's a pain (I don't mind doing it in a follow up)